ext/include/opentelemetry/ext/http/client/curl/http_client_curl.h

```diff
@@ -153,6 +153,33 @@ class Session : public http_client::Session
     });
   }
 
+  virtual std::unique_ptr<http_client::Response> SendRequestSync(
```

Comment:
I believe it would be much cleaner if you operate only on `Response` object rather than on `SessionState`. Let me elaborate - right now you have two possible states:
- success - in this case you return `Response`; or
- falure - in this case you force the user to test if `Response` ptr is nullptr, and then inspect the `SessionState` parameter passed by reference.. which is not really `Session` state (because `Session` may have multiple requests), but rather `Request` state.

So since in reality it is `Request` state, can you have this aggregated on `Response` object - making sure that you always return `Response`. It may be successful response or empty response, failed response, with an indication on `Response` object that it's a failed response. Then you avoid the nullptr entirely, you avoid checking on two separate objects.


Could you please clarify - if the original intent of `SessionManager` was to support more than one concurrent request at a time? If so, then `SessionState` is a misleading name for a synchronous request. Session may have more than one request, one failed, one successful, and `SessionState` name does not represent the actual `SessionManager` Session state.

Reply:
`Response` object represents HTTP Response containing - body, header and code. Any error during http handshake are returned in `SessionState`. This is to keep it consistent with how we achieve it in async requests:
```
EventHandler::OnResponse(Response &);
EventHandler::OnError(SessionState);
Session::SendRequest(EventHandler &); // async
unique_ptr<Request> Session::SendRequestSync(SessionState &); // sync
```

`Session` is more of logical entity, and doesn't represent multiple ongoing http requests.